### PR TITLE
[SPARK-36682][CORE][TEST] Add Hadoop sequence file test for different Hadoop codecs

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

This patch proposes to add e2e tests for using Hadoop codecs to write sequence files.

### Why are the changes needed?

To improve test coverage.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Added tests.

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/FileSuite.scala [hunk 1]
```diff
@@ -113,25 +113,33 @@ class FileSuite extends SparkFunSuite with LocalSparkContext {
...
 
-    assert(compressedFile.length < normalFile.length)
+      assert(compressedFile.length < normalFile.length)
+    }
+  }
+
+  // Hadoop "gzip" codec doesn't support sequence file yet.
+  // Hadoop "zstd" codecs needs native library installed.
+  Seq((new DefaultCodec(), "default"), (new BZip2Codec(), "bzip2"), (new GzipCodec(), "gzip"),
+    (new Lz4Codec(), "lz4"), (new SnappyCodec, "snappy")).foreach { case (codec, codecName) =>
```
Comment (viirya) [inline core/src/test/scala/org/apache/spark/FileSuite.scala:141]: lz4 codecs currently fails due to SPARK-36669.
snappy codec currently fails due to SPARK-36681.

Comment (dongjoon-hyun) [reply core/src/test/scala/org/apache/spark/FileSuite.scala:141]: Thank you for filing JIRAs.

**Comment by viirya** [comment]
We have dedicated JIRAs for snappy and lz4 issues. We already verified the issues in above CI test, going to remove them first to make test passed.

**Comment by viirya** [comment]
cc @sunchao @cloud-fan @dbtsai too

**Review by dbtsai — APPROVED** [review]
LGTM. Thanks!

**Review by sunchao — APPROVED** [review]
LGTM

File: core/src/test/scala/org/apache/spark/FileSuite.scala [hunk 2]
```diff
@@ -113,25 +113,34 @@ class FileSuite extends SparkFunSuite with LocalSparkContext {
...
+      val compressedFile = new File(compressedOutputDir, "part-00000" + codec.getDefaultExtension)
+      val compressedContent = sc.sequenceFile[String, String](compressedOutputDir).collect
+      assert(compressedContent === Array.fill(100)(("abc", "abc")))
 
-    assert(compressedFile.length < normalFile.length)
+      assert(compressedFile.length < normalFile.length)
+    }
+  }
+
+  // Hadoop "gzip" codec doesn't support sequence file yet.
```
Comment (dbtsai) [inline core/src/test/scala/org/apache/spark/FileSuite.scala:138]: `Hadoop "gzip" codec requires native library installed for gzip compressed seq file.`

File: core/src/test/scala/org/apache/spark/FileSuite.scala [hunk 3]
```diff
@@ -113,25 +113,34 @@ class FileSuite extends SparkFunSuite with LocalSparkContext {
...
+  // Hadoop "zstd" codec needs native library installed.
```
Comment (dbtsai) [inline core/src/test/scala/org/apache/spark/FileSuite.scala:139]: zstd is the same as gzip before your gzip codec is released in hadoop. Maybe just say

`Hadoop "gzip" and "zstd" codecs requires native library installed for sequence files`

Comment (viirya) [reply core/src/test/scala/org/apache/spark/FileSuite.scala]: Thanks. Revised as suggested.

**Comment by viirya** [comment]
The last commit is comment doc only. Thanks for reviewing. Merging to master/3.2.

**Comment by dongjoon-hyun** [comment]
+1, late LGTM.

**Comment by cloud-fan** [comment]
late LGTM

**Pull request closed without merging**